tutorials/bert_e2e.py

```diff
@@ -0,0 +1,206 @@
+    and output the result in text format for model like QA.
+
+    Args:
+        model_name (str): Which model to export in hugging-face, it determinate tokenizer and onnx model backbone.
+        input_model_file (Path): The onnx model needed to be saved/cached, if not provided, will export from hugging-face.
+        output_model_file (Path): where to save the final onnx model.
+    """
+    tokenizer, bert_onnx_model = export_backbone(model_name, input_model_file)
+    if not hasattr(tokenizer, "vocab_file"):
+        vocab_file = bert_onnx_model.parent / "vocab.txt"
+        import json
+        with open(str(vocab_file),'w') as f:
+            f.write(json.dumps(tokenizer.vocab))
+    else:
+        vocab_file = tokenizer.vocab_file
+    add_ppp.transformers_and_bert(bert_onnx_model, output_model_file, model_name, vocab_file,add_debug_before_postprocessing=True)
+    
+
+def verify_results_for_e2e_model(output_model_file: Path, model_name: str, input_bert_model: Path = None):
+    """
+    Args:
+        output_model_file: the onnx model which finalized and needs to be verified
+        model_name: the huggingface model name
+        input_bert_model: the onnx model which is generated by huggingface or user provide
+    """
+    tokenizer, hg_model, _, text = get_tokenizer_and_model_from_huggingface(model_name)
+    encoded_input = tokenizer(*text, return_tensors="pt")
+    transformers.set_seed(42)
+
+    session_options = onnxruntime.SessionOptions()
+
+    if input_bert_model and input_bert_model.exists():
+        session = onnxruntime.InferenceSession(
+            str(input_bert_model.resolve(strict=True)), providers=["CPUExecutionProvider"]
+        )
+        inputs = {key: value.detach().numpy() for key, value in encoded_input.items()}
+
+        ref_outputs = session.run([i.name for i in session.get_outputs()], inputs)
+        ref_map_out = {i.name: ref_outputs[idx] for idx, i in enumerate(session.get_outputs())}
+    else:
+        outs = hg_model(**encoded_input)
+        ref_outputs = [out.detach().numpy() for out in list(outs.values())]
+        ref_map_out = {i: ref_outputs[idx] for idx, i in enumerate(outs.keys())}
+
+    # Load tokenizer op
+    session_options.register_custom_ops_library(onnxruntime_extensions.get_library_path())
+
+    session = onnxruntime.InferenceSession(
+        str(output_model_file.resolve(strict=True)), session_options, providers=["CPUExecutionProvider"]
+    )
+
+    # build input, QA task has 2 inputs
+    if len(text) == 2:
+        inputs = dict(
+            inputs=np.array([[i] for i in text]),
+        )
+    else:
+        inputs = dict(
+            inputs=np.array([i for i in text]),
+        )
+    real_outputs = session.run([i.name for i in session.get_outputs()], inputs)
+    matched_idx = [i for i, o in enumerate(session.get_outputs()) if list(ref_map_out.keys())[0] in o.name][0]
```

Comment:
Can we simplify and provide the list of output names in the call to 'run'? ORT will return the outputs in that order, so if we do that in both calls to 'run' the order will match.

Reply:
Sure.